mavros/scripts/mavsys

```diff
@@ -20,7 +20,11 @@ from mavros_msgs.srv import SetMode, StreamRate, StreamRateRequest
 
 
 def do_mode(args):
-    base_mode = 0
+    # Setting to MAV_MODE_FLAG_CUSTOM_MODE_ENABLED is mandatory, so to change to
+    # a custom mode/submode.
+    # NOTE: In order to change to AUTO.MISSION on PX4 Pro, you need to disable
+    # datalink failsafe, which can be done by setting NAV_DLL_ACT to 0.
+    base_mode = 1 # 1 = MAV_MODE_FLAG_CUSTOM_MODE_ENABLED
```

Comment:
Please read the code *before* edit!

[sys_status.cpp 931-945](https://github.com/mavlink/mavros/blob/master/mavros/src/plugins/sys_status.cpp#L931-L945)
```cpp
		if (req.custom_mode != "") {
			if (!m_uas->cmode_from_str(req.custom_mode, custom_mode)) {
				res.mode_sent = false;
				return true;
			}

			/**
			 * @note That call may trigger unexpected arming change because
			 *       base_mode arming flag state based on previous HEARTBEAT
			 *       message value.
			 */
			base_mode |= (m_uas->get_armed()) ? enum_value(MAV_MODE_FLAG::SAFETY_ARMED) : 0;
			base_mode |= (m_uas->get_hil_state()) ? enum_value(MAV_MODE_FLAG::HIL_ENABLED) : 0;
			base_mode |= enum_value(MAV_MODE_FLAG::CUSTOM_MODE_ENABLED);
		}
```

Reply:
So you have to explain me why without setting the base_mode to 1 in mavsys it does not change to flight submodes. From my evaluation, what you have above is not doing what is supposed.

Reply:
The original issue that I faced was https://github.com/mavlink/mavros/issues/807. I was unable to switch to MISSION mode after uploading missions. But whereas I was able to do so in QGC.
So, as me & @TSC21  went on experimenting  the necessary changes, we had set `base_mode=1` & unset of `NAV_DLL_ACT`. In reality only later change is required, not `base_mode=1`.  
Switching to LOITER mode too wasn't happening. This was also resolved by doing unset of `NAV_RCL_ACT`. 
So, the conclusion is the mode switch to AUTO.MISSION & AUTO.LOITER will happen by setting parameters `NAV_DLL_ACT` & `NAV_RCL_ACT` to `0`. It is verified by using both C++ and `mavsys` commands.
```
$ rosrun mavros mavsys mode -c AUTO.MISSION
Mode changed
```
Thanks @TSC21 for suggesting param set commands which resolved my issue.
--Shakthi